Ryujinx.Graphics.Vulkan/TextureStorage.cs

```diff
@@ -440,31 +442,39 @@ public void SetModification(AccessFlags accessFlags, PipelineStageFlags stage)
             _lastModificationStage = stage;
         }
 
-        public void InsertBarrier(CommandBufferScoped cbs, AccessFlags dstAccessFlags, PipelineStageFlags dstStageFlags)
+        public void InsertReadToWriteBarrier(CommandBufferScoped cbs, AccessFlags dstAccessFlags, PipelineStageFlags dstStageFlags)
         {
-            if (_lastModificationAccess != AccessFlags.NoneKhr)
+            if (_lastReadAccess != AccessFlags.NoneKhr)
             {
-                ImageAspectFlags aspectFlags;
+                ImageAspectFlags aspectFlags = EnumConversion.ConvertAspectFlags(Info.Format);
```

Comment:
```suggestion
                ImageAspectFlags aspectFlags = Info.Format.ConvertAspectFlags();
```
Since it's a extension method, I think you can just do that (same for the other one).
And would be nice to fix that spacing on that method as I commented before:
```cs
GAL.Format.S8UintD24Unorm    => ImageAspectFlags.DepthBit | ImageAspectFlags.StencilBit,
```